### PR TITLE
bump: Scala 2.13.10 and 2.12.17; Javadoc links

### PR DESCRIPTION
**Pull request by ennru**
...and remove some link checking left-overs.

### CODE REVIEW

File: s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala [hunk 1]
```diff
@@ -552,7 +552,7 @@ trait S3IntegrationSpec
       _ <- akka.pattern.after(25.seconds)(Future {
         sharedKillSwitch.abort(AbortException)
       })
-      _ <- multiPartUpload.recover {
+      _ <- multiPartUpload.recover[Any] {
```
Comment (He-Pin) [inline s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala:555]: val defaultScalaOptions = "-Wconf:cat=unused-nowarn:s,cat=lint-infer-any:s,any:e"

File: amqp/src/test/scala/akka/stream/alpakka/amqp/scaladsl/AmqpGraphStageLogicConnectionShutdownSpec.scala [hunk 2]
```diff
@@ -89,7 +89,7 @@ class AmqpGraphStageLogicConnectionShutdownSpec
 
     Future
       .traverse(input)(in => Source.single(ByteString(in)).runWith(amqpSink))
-      .recover {
+      .recover[Object] {
```
Comment (patriknw) [inline amqp/src/test/scala/akka/stream/alpakka/amqp/scaladsl/AmqpGraphStageLogicConnectionShutdownSpec.scala:92]: `AnyRef`

File: influxdb/src/test/scala/docs/scaladsl/InfluxDbSourceSpec.scala [hunk 3]
```diff
@@ -61,7 +61,7 @@ class InfluxDbSourceSpec
     val query = new Query("SELECT man() FROM invalid", DatabaseName);
 
     val result = InfluxDbSource(influxDB, query) //.runWith(Sink.seq)
-      .recover {
+      .recover[Object] {
```
Comment (patriknw) [inline influxdb/src/test/scala/docs/scaladsl/InfluxDbSourceSpec.scala:64]: `AnyRef` in several places

File: scripts/link-validator.conf [hunk 4]
```diff
@@ -39,7 +35,7 @@ site-link-validator {
     "http://www.thedevpiece.com/"
     # genereated by @apidoc
     "http://pravega.io/"
-    "http://www.scala-lang.org/api/2.13.8/scala/concurrent/Future.html"
-    "http://www.scala-lang.org/api/2.13.8/scala/util/Try.html"
+    "http://www.scala-lang.org/api/2.13.10/scala/concurrent/Future.html"
+    "http://www.scala-lang.org/api/2.13.10/scala/util/Try.html"
```
Comment (patriknw) [inline scripts/link-validator.conf:39]: can these be https://www.scala-lang.org/api/2.13.x/scala/concurrent/Future.html ?

Comment (ennru) [reply scripts/link-validator.conf:39]: Sadly not. The generated links contain the exact version.

**Comment by ennru** [comment]
I updated the mocks in Google Pub/Sub so they don't always fail.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**